dragnet/compat.py

```diff
@@ -8,20 +8,259 @@
+    """
+    block.text = bytes_cast(block.text, **kwargs)                    if include_text        else block.text
+    block.link_tokens = bytes_list_cast(block.link_tokens, **kwargs) if include_link_tokens else block.link_tokens
+    block.css = bytes_dict_cast(block.css, **kwargs)                 if include_css         else block.css
+    block.features = bytes_dict_cast(block.features, **kwargs)       if include_features    else block.features
+    return block
+
+
+def str_block_list_cast(blocks, **kwargs):
+    """
+    Converts any bytes-like items in input lxml.Blocks to string-like values, with respect to python version
+
+    Parameters
+    ----------
+    blocks : list[lxml.Block]
+        any bytes-like objects contained in the block object will be converted to a string (assumed utf-8 encoding)
+    kwargs:
+        include_text : bool, default=True
+            if True, cast text to a string, else ignore
+        include_link_tokens : bool, default=True
+            if True, cast link_tokens to a string, else ignore
+        include_css : bool, default=True
+            if True, cast css to a string, else ignore
+        include_features : bool, default=True
+            if True, cast features to a string, else ignore
+        encoding: str, default: 'utf-8'
+            encoding to be used when decoding bytes
+    """
+    return [str_block_cast(block, **kwargs) for block in blocks]
+
+
+def bytes_block_list_cast(blocks, **kwargs):
+    """
+    Converts any string-like items in input lxml.Blocks to bytes-like values, with respect to python version
+
+    Parameters
+    ----------
+    blocks : list[lxml.Block]
+        any string-like objects contained in the block object will be converted to bytes (assumed utf-8 encoding)
+    kwargs:
+        include_text : bool, default=True
+            if True, cast text to bytes, else ignore
+        include_link_tokens : bool, default=True
+            if True, cast link_tokens to bytes, else ignore
+        include_css : bool, default=True
+            if True, cast css to bytes, else ignore
+        include_features : bool, default=True
+            if True, cast features to bytes, else ignore
+        encoding: str, default: 'utf-8'
+            encoding to be used when decoding bytes
+    """
+    return [bytes_block_cast(block, **kwargs) for block in blocks]
+
+
 # scikit-learn version compatibility
 from sklearn import __version__ as sklearn_version
 
+if sklearn_version < '0.18.0':
+    from sklearn.cross_validation import train_test_split
+    from sklearn.grid_search import GridSearchCV
+else:
+    from sklearn.model_selection import train_test_split
+    from sklearn.model_selection import GridSearchCV
+
+# generate model paths
 if '0.15.2' <= sklearn_version <= '0.17.1':
-    sklearn_path = 'sklearn_0.15.2_0.17.1'
+    if PY2:
+        model_path = 'py2_sklearn_0.15.2_0.17.1'
+    else:
+        model_path = 'py3_sklearn_0.15.2_0.17.1'
```

Comment:
This model path does not exist. Causing error if sklearn version is from 0.152 to 0.17.1 while using python3

Reply:
Sorry about that, I missed this file :) Added/pushed it